ca/adminClient.go

```diff
@@ -1101,6 +1101,103 @@ retry:
 	return nil
 }
 
+func (c *AdminClient) CreateProvisionerWebhook(provisionerName string, wh *linkedca.Webhook) (*linkedca.Webhook, error) {
+	var retried bool
+	body, err := protojson.Marshal(wh)
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling request: %w", err)
+	}
+	u := c.endpoint.ResolveReference(&url.URL{Path: path.Join(adminURLPrefix, "provisioners", provisionerName, "webhooks")})
+	tok, err := c.generateAdminToken(u)
+	if err != nil {
+		return nil, fmt.Errorf("error generating admin token: %w", err)
+	}
+	req, err := http.NewRequest(http.MethodPost, u.String(), bytes.NewReader(body))
+	if err != nil {
+		return nil, fmt.Errorf("creating POST %s request failed: %w", u, err)
+	}
+	req.Header.Add("Authorization", tok)
+retry:
```

Comment:
Make sure the `bytes.NewReader(body)` doesn't get consumed if there's a retry. We should move this retry and the ones below if it gets consumed.

I commented on something similar before, in that method, the retry can also go before the `Do(req)` if we can reuse the same request.

Reply:
58be425